src/main/java/org/spongepowered/api/attribute/Attribute.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * This file is part of Sponge, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered.org <http://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.attribute;
+
+import org.spongepowered.api.text.message.Message;
+
+import com.google.common.base.Predicate;
+
+/**
+ * Represents an type of attribute that can be applied to a
+ * {@link AttributeHolder}.
+ */
+public interface Attribute {
+
+    /**
+     * Gets the ID of this attribute as a string.
+     * 
+     * @return The ID of this attribute as a string
+     */
+    String getId();
+
+    /**
+     * Gets the minimum value for this attribute.
+     * 
+     * @return The minimum value for this attribute
+     */
+    double getMinimum();
+
+    /**
+     * Gets the maximum value for this attribute.
+     * 
+     * @return The maximum value for this attribute
+     */
+    double getMaximum();
+
+    /**
+     * Gets the default value for this attribute.
+     * 
+     * @return The default value for this attribute
+     */
+    double getDefaultBase();
+
+    /**
+     * Gets the target of this attribute.
+     * 
+     * @return The target of this attribute
+     */
+    Predicate<AttributeHolder> getTarget();
```

Comment:
`Targets` is better than `Target`, but look at what this method actualy does.
The target is not really a targets its more an `isApplicable(AttributeHolder)`, and then @gratimax would be right that it can be checked for `EntityType`.

So i suggest changing it to `boolean isApplicable(AttributeHolder)` or `boolean isApplicable(EntityType)`.